CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeNetworkEngine.cs

```diff
@@ -334,6 +340,42 @@ private void ApplyNetworkEngineRequestConfiguration(UnityNativeRequest request)
             }
         }
 
+        private Dictionary<string, string> AddARP(Dictionary<string, string> headers)
+        {
+            string arpNamespaceKey = string.Format(UnityNativeConstants.Network.ARP_NAMESPACE_KEY,
+                _coreState.AccountInfo.AccountId, _coreState.DeviceInfo.DeviceId);
+            var arpJson = _preferenceManager.GetString(arpNamespaceKey, string.Empty);
+            if (string.IsNullOrEmpty(arpJson))
+                return headers;
+
+            Dictionary<string, object> arpDictionary = Json.Deserialize(arpJson) as Dictionary<string, object>;
+            if (arpDictionary == null || arpDictionary.Count == 0)
+                return headers;
+
+            var keysToRemove = new List<string>();
+    
+            foreach (var param in arpDictionary)
+            {
+                if (param.Value is string strValue && strValue.Length > 100)
+                {
+                    keysToRemove.Add(param.Key);
+                }
+                else if (param.Value is int intValue && intValue == -1)
+                {
+                    keysToRemove.Add(param.Key);
+                }
+            }
+
+            foreach (var key in keysToRemove)
+            {
+                arpDictionary.Remove(key);
+            }
+
+            headers[UnityNativeConstants.Network.ARP_KEY] = Json.Serialize(arpDictionary);
```

Comment:
Does the BE accepts the serialized ARP like this?

Reply:
UnityWebRequests can only be Dictionary<string,string>. I will verify ARP updates.